test/integration-test/TestIIRDeserializer.cpp

```diff
@@ -0,0 +1,598 @@
+  target->getMetaData().addStencilCallStmt(stencilCallDeclStmt, stencilID);
+
+  auto stencilCallStatement = std::make_shared<Statement>(stencilCallDeclStmt, nullptr);
+  target->getIIR()->getControlFlowDescriptor().insertStmt(stencilCallStatement);
+
+  ///////////////// Generation of the Metadata
+
+  target->getMetaData().insertExprToAccessID(lhsTmp, tmpFieldID);
+  target->getMetaData().insertExprToAccessID(rhsInT1, inFieldID);
+  target->getMetaData().insertExprToAccessID(rhsInT2, inFieldID);
+  target->getMetaData().insertExprToAccessID(rhsInT3, inFieldID);
+  target->getMetaData().insertExprToAccessID(rhsInT4, inFieldID);
+
+  target->getMetaData().insertExprToAccessID(lhsOut, outFieldID);
+  target->getMetaData().insertExprToAccessID(rhsTmpT1, tmpFieldID);
+  target->getMetaData().insertExprToAccessID(rhsTmpT2, tmpFieldID);
+  target->getMetaData().insertExprToAccessID(rhsTmpT3, tmpFieldID);
+  target->getMetaData().insertExprToAccessID(rhsTmpT4, tmpFieldID);
+
+  target->getMetaData().setStencilname("generated");
+
+  for(const auto& MS : iterateIIROver<iir::MultiStage>(*(target->getIIR()))) {
+    MS->update(iir::NodeUpdateType::levelAndTreeAbove);
+  }
+  // Iterate all statements (top -> bottom)
+  for(const auto& stagePtr : iterateIIROver<iir::Stage>(*(target->getIIR()))) {
+    iir::Stage& stage = *stagePtr;
+    for(const auto& doMethod : stage.getChildren()) {
+      doMethod->update(iir::NodeUpdateType::level);
+    }
+    stage.update(iir::NodeUpdateType::level);
+  }
+  for(const auto& MSPtr : iterateIIROver<iir::Stage>(*(target->getIIR()))) {
+    MSPtr->update(iir::NodeUpdateType::levelAndTreeAbove);
+  }
+}
+
+void readIIRFromFile(OptimizerContext &optimizer, std::shared_ptr<iir::StencilInstantiation>& target, std::string fname) {
+  target =
+      IIRSerializer::deserialize(fname, &optimizer, IIRSerializer::SK_Json);
+
+  //this is whats actually to be tested.
+  optimizer.restoreIIR("<restored>", target);
+}
+
+void compareIIRs(std::shared_ptr<iir::StencilInstantiation> lhs, std::shared_ptr<iir::StencilInstantiation> rhs) {
+  //first compare the (structure of the) iirs, this is a precondition before we can actually check the metadata / derived info
+  compareIIRstructures(lhs->getIIR().get(), rhs->getIIR().get());
+
+  //then we compare the meta data
+  compareMetaData(lhs->getMetaData(), rhs->getMetaData());
+
+  //and finally the derived info 
+  compareDerivedInformation(lhs->getIIR().get(), rhs->getIIR().get());  
+}
+
+TEST(IIRDeserializerTest, CopyStencil) {
+  Options compileOptions;
+  OptimizerContext::OptimizerContextOptions optimizerOptions;
+  DawnCompiler compiler(&compileOptions);   
+  OptimizerContext optimizer(compiler.getDiagnostics(), optimizerOptions, std::make_shared<dawn::SIR>());
+
+  //read IIR from file
+  std::shared_ptr<iir::StencilInstantiation> copy_stencil_from_file = std::make_shared<iir::StencilInstantiation>(
```

Comment:
```suggestion
  auto copy_stencil_from_file = std::make_shared<iir::StencilInstantiation>(
```
Very natural place to put auto (also below).